### ISSUE TITLE
Parallel BrowserStack tests

### ISSUE THREAD

**Issue by raineorshine** [cybersemics/em#1475]
An issue with parallel Just tests in BrowserStack was noted in #1470. The current solution does not scale beyond 3 test files. A better solution would be to limit the concurrency of the Jest tests. I suggest `--maxWorkers` at the end, but it has yet to be tested.

> @raineorshine I was observing that the browser stack tests failed intermittently during github checks. I also tested it from my local machine and it sometimes failed intermittently. I looked into it and came to a convincing reason for it.
> 
> So this is how browser stack tests are running.
> 
> * Jest environment is started.
> * It creates a browser stack local testing connection with a local identifier. A local identifier is used to identify the connection we make to the browser stack.
> * webdriverio is used to connect to the browser stack to run the tests.
> 
> However, the problem was that the jest ran three tests in parallel. But we for each test we created a browser stack local testing connection with the same local identifier. That meant if one of the tests completes and stops the connection with that local identifier, that connection would probably close for others too. So that caused some tests to fail due to connection errors.
> 
> So now the connection is started with the unique local identifier for every test that is running in parallel. I tested it by running the tests multiple times. And it seems to be working without failure.
> 
> However, we need to make sure we don't max out the parallel threads provided by BrowserStack. For now, we have only three tests, so it will work without problem.
> 
> Please let me know what you think about it.

> > How can we limit the number of parallel threads to 3 if we add additional iOS test files?
> 
> I am not sure about this. For now, the number of threads is equal to the number of ios test files. The limit for the basic plan is 5 parallel threads. To overcome this we can either run tests in a sequence that will increase the total test time. Or we need to find a way to instantiate a connection once before e2e tests run, share it among all the test files and close it after all the tests are executed. I have little experience on BrowserStack so there could possibly be other ways too.

> > Cool. Thanks for sharing your ideas.
> > Ideally we could limit concurrent tests, maybe at the web driver level.
> 
> I believe that should be done on jest level. Since it runs tests in parallel. If we could configure jest to use a maximum of 5 test files in parallel, it could be the best solution to this problem. I am not sure if we have that option in jest. However, we can always resort to running tests in sequence using `--runInBand` flag in jest.

> > Could you try `--maxWorkers`?
> > [jestjs.io/docs/cli#--maxworkersnumstring](https://jestjs.io/docs/cli#--maxworkersnumstring)
> 
> I'll create a separate issue for this.
